src/index.ts

```diff
@@ -76,4 +79,62 @@ app.post("/logout", async(c) => {
   return new Response();
 });
 
+app.post("/attendance", async(c) => {
+  /*ネットワークの確認を行う*/
+  const allowPattern = new RegExp(c.env.ALLOW_IP_REGEX);
+  if ( !allowPattern.test(c.req.header("cf-connectiong-ip") ?? "") ) { //"??":c.req.header(cf-connectiong-ip")でnullが返ってきたときに""として扱う演算子
+    return c.text("Forbidden", 403);
+  }
+
+  /*学生の認証を行う*/
+  //sessionがcookieあるか確認
+  const honoSession = c.get('session');
+  const session = honoSession.get("login") as Session | null;
+  if ( !session ){
+    return c.text("Unauthorized", 401);
+  }
+
+  /*まとめてクエリを実行する*/
+  const now = new Date();
+  const [ existResult, accountEntry, attendanceBoardEntry ] = await c.env.DB.batch([
+    c.env.DB.prepare("SELECT * FROM session WHERE id = ?").bind(session.id),
+    c.env.DB.prepare("SELECT * FROM account WHERE role = 'STUDENT' AND id = ?").bind(session.account.id),
+    c.env.DB.prepare("SELECT id FROM attendance_board WHERE start_from  <= ?1 AND seconds_from_be_late_to_end >= ?2").bind(now, now),
```

Comment:
```suggestion
    c.env.DB.prepare("SELECT id FROM attendance_board WHERE start_from  <= ?1 AND seconds_from_be_late_to_end >= ?1").bind(now),
```
同じ値を当てはめるのであれば `?1` を 2 回使えばよいかと思います.